sqlglot/dialects/starrocks.py

```diff
@@ -35,7 +35,10 @@ def _parse_unnest(self, with_alias: bool = True) -> t.Optional[exp.Unnest]:
             if unnest:
                 alias = unnest.args.get("alias")
 
-                if alias and not alias.args.get("columns"):
+                if not alias:
+                    alias = exp.TableAlias(columns=[exp.to_identifier("unnest")])
+                    unnest.set("alias", alias)
```

Comment:
Is this tested? What does generation of `UNNEST` without an alias look like in Starrocks after this change?

Reply:
@hellozepp I think my recommendation of removing the table alias might have broken the roundtrip here, sorry for that. We can test the roundtrip in the first new `test_starrocks.py` test case you added, i.e:

```
        self.validate_all(
            "SELECT student, score, unnest FROM tests CROSS JOIN LATERAL UNNEST(scores)",
            write={
                "spark": "SELECT student, score, unnest FROM tests LATERAL VIEW EXPLODE(scores) AS unnest",
                "starrocks": "SELECT student, score, unnest FROM tests CROSS JOIN LATERAL UNNEST(scores) AS unnest(unnest)",
            },
        )
```

As you suggested, Starrocks seems to name both the `UNNEST` table and it's output column as `unnest`. Here's the `DESCRIBE` result for a similar query:

```
StarRocks > describe with t as (select 1 as col, [1, 2, 3] as col2) select unnest from t cross join unnest(col2);
+--------------------------------+
| Explain String                 |
+--------------------------------+
| PLAN FRAGMENT 0                |
|  OUTPUT EXPRS:4: unnest        |
|   PARTITION: UNPARTITIONED     |
|                                |
|   RESULT SINK                  |
|                                |
|   2:TableValueFunction         |
|   |  tableFunctionName: unnest |
|   |  columns: [unnest]         |
|   |  returnTypes: [TINYINT]    |
|   |                            |
|   1:Project                    |
|   |  <slot 3> : [1,2,3]        |
|   |                            |
|   0:UNION                      |
|      constant exprs:           |
|          NULL                  |
+--------------------------------+
17 rows in set (0.02 sec)
```

Reply:
@VaggelisD Oh, I understand. Here we just need to avoid setting the default unnest alias when table alias exists. I revert the code back.

Reply:
@VaggelisD 
![image](https://github.com/user-attachments/assets/78f2a080-e3dc-4565-83c6-612b5c358568)